pkg/rangemap/range.go

```diff
@@ -0,0 +1,40 @@
+package rangemap
+
+import (
+	"bytes"
+	"errors"
+	"github.com/anujga/dstk/pkg/core"
+	"github.com/google/btree"
+)
+
+var ErrInvalidRange = errors.New("invalid range")
+
+type Range interface {
+	Start() core.KeyT
+	End() core.KeyT
+}
+
+type rangeItem struct {
```

Comment:
Why is the wrapper required

Reply:
rangeItem encapsulates logic to see if a key belongs to a range, if a range precedes another

Reply:
That logic can work on the interface just the same